core/primitives/src/views.rs

```diff
@@ -1017,6 +1017,8 @@ pub struct ExecutionOutcomeView {
     pub executor_id: AccountId,
     /// Execution status. Contains the result in case of successful execution.
     pub status: ExecutionStatusView,
+    /// Execution metadata, versioned
+    pub metadata: ExecutionMetadata,
```

Comment:
Please, skip serde serialization and deserialization for now, so we don't leak this field to the JSON RPC users just yet.

Reply:
OK! Note that we'll need this eventually because the final purpose is to expose profile data via RPC.

Reply:
I think we'd actually want to use a different struct for JSON RPC. versioned enum is a borsh-friendly format. For JSON we probably want just a struct here. It's funny how borsh and json are duals in a sense that the most natural forward compat field in JSON is a struct to which you can add fields, while in borsh it's an enum to which you can add variants.

Reply:
@matklad That is totally on my radar of refactoring JSON RPC! :+1: